### PR TITLE
Fix algorithm to do >=, fix tests for correction behaviour and reduce to one implementation

### PR DESCRIPTION
**Pull request by uhthomas**
Hey! I noticed the current staling sort implementations written in Go were incorrect. One of them did `>` rather than `>=` meaning `2, 2, 2` would be truncated to just `2` and another test was just somehow plain wrong.

### CODE REVIEW

File: go/stalin/stalin_test.go [hunk 1]
```diff
@@ -1,59 +0,0 @@
...
-		}
-
-		remaining := list[n:]
-		if !reflect.DeepEqual(remaining, make([]int, len(remaining))) {
-			t.Errorf("%v: remaining not zero: %v", gold.Got, remaining)
-		}
-	}
-}
-
-func BenchmarkSort(b *testing.B) {
```
Comment (gustavo-depaula) [inline go/stalin/stalin_test.go]: can we migrate this benchmark?

Comment (uhthomas) [reply go/stalin/stalin_test.go]: Done! :) If you're interested in the difference
```
$ benchcmp old.txt new.txt                  
benchmark                        old ns/op     new ns/op     delta
BenchmarkSortInts/stalin-8       17.1          14.0          -18.13%
```

**Comment by uhthomas** [comment]
@gustavo-depaula Is this good to merge?

**Comment by gustavo-depaula** [comment]
great work!
thanks for your contribution! 🚀

**Pull request merged**